### PR TITLE
Update md5-used-as-password messages to recommend proper password hashing functions.

### LINKED ISSUE #1609: md5-used-as-password incorrectly recommends SHA256 as alternative
* [semgrep-rules/md5-used-as-password.yaml at develop · returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/blob/develop/php/lang/security/md5-used-as-password.yaml)
* [semgrep-rules/md5-used-as-password.yaml at develop · returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/blob/develop/ruby/lang/security/md5-used-as-password.yaml)
* [semgrep-rules/md5-used-as-password.yaml at develop · returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/blob/develop/go/lang/security/audit/md5-used-as-password.yaml)
* [semgrep-rules/md5-used-as-password.yaml at develop · returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/blob/develop/java/lang/security/audit/md5-used-as-password.yaml)
* [semgrep-rules/md5-used-as-password.yaml at develop · returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/blob/develop/python/lang/security/audit/md5-used-as-password.yaml)
* [semgrep-rules/md5-used-as-password.yaml at develop · returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/blob/develop/javascript/lang/security/audit/md5-used-as-password.yaml)

> It looks like MD5 is used as a password hash. MD5 is not considered a secure password hash because it can be cracked by an attacker in a short    amount of time. Use SHA-256 or higher for password hashes (for example: `hash('sha256', ...)`)

The recommendation, *Use SHA-256 or higher for password hashes*, is incorrect. 
SHA256 is also not a proper password hashing function, as it is too fast. Acceptable hash functions are PBKDF2, bcrypt, scrypt, argon2. Most languages/frameworks have a supported password hashing function, we should recommend that. E.g. for PHP there's [password_hash](https://www.php.net/password_hash).

### PR DESCRIPTION
**Pull request by minusworld**
Fixes https://github.com/returntocorp/semgrep-rules/issues/1609

### CODE REVIEW

**Review by inkz — APPROVED** [review]
:clap:

File: python/lang/security/audit/md5-used-as-password.yaml [hunk 1]
```diff
@@ -4,7 +4,8 @@ rules:
   message: >-
     It looks like MD5 is used as a password hash. MD5 is not considered a
     secure password hash because it can be cracked by an attacker in a short
-    amount of time. Use SHA-256 or higher for password hashes (for example: `hashlib.sha256(...)`)
+    amount of time. Use a suitable password hashing function such as scrypt.
+    You can use `hashlib.script`.
```
Comment (Sjord) [inline python/lang/security/audit/md5-used-as-password.yaml:8]: `script` should be `scrypt` here.

Comment (inkz) [reply python/lang/security/audit/md5-used-as-password.yaml:8]: @Sjord  Thanks a lot :pray:

Comment (inkz) [reply python/lang/security/audit/md5-used-as-password.yaml:8]: PR: https://github.com/returntocorp/semgrep-rules/pull/1653

**Pull request merged**